test/CMakeLists.txt

```diff
@@ -0,0 +1,149 @@
+add_subdirectory(includes)
+
+get_property(
+  TEST_INCLUDE_DIRS DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  PROPERTY INCLUDE_DIRECTORIES
+)
+
+# When running tests from 'ninja' we need to use the
+# console pool: to do so we need to use the USES_TERMINAL
+# option, but this is only available in CMake 3.2
+set(TEST_TARGET_ARGS)
+list(APPEND TEST_TARGET_ARGS "USES_TERMINAL")
+
+set(UNITTEST_PREREQS nvim nvim-test unittest-headers)
+set(FUNCTIONALTEST_PREREQS nvim printenv-test printargs-test shell-test streams-test tty-test ${GENERATED_HELP_TAGS})
+set(BENCHMARK_PREREQS nvim tty-test)
+
+# Useful for automated build systems, if they want to manually run the tests.
+add_custom_target(
+  unittest-prereqs
+  DEPENDS ${UNITTEST_PREREQS}
+)
+set_target_properties(unittest-prereqs PROPERTIES FOLDER test)
+
+add_custom_target(
+  functionaltest-prereqs
+  DEPENDS ${FUNCTIONALTEST_PREREQS}
+)
+
+add_custom_target(
+  benchmark-prereqs
+  DEPENDS ${BENCHMARK_PREREQS}
+)
+
+check_lua_module(${LUA_PRG} "ffi" LUA_HAS_FFI)
+if(LUA_HAS_FFI)
+  add_custom_target(
+    unittest
+    COMMAND ${CMAKE_COMMAND}
+    -DBUSTED_PRG=${BUSTED_PRG}
+    -DLUA_PRG=${LUA_PRG}
+    -DWORKING_DIR=${CMAKE_CURRENT_SOURCE_DIR}
+    -DBUSTED_OUTPUT_TYPE=${BUSTED_OUTPUT_TYPE}
+    -DTEST_DIR=${CMAKE_CURRENT_SOURCE_DIR}/test
+    -DBUILD_DIR=${CMAKE_BINARY_DIR}
+    -DTEST_TYPE=unit
+    -P ${PROJECT_SOURCE_DIR}/cmake/RunTests.cmake
+    DEPENDS ${UNITTEST_PREREQS}
```

Comment:
Shouldn't you just depend on the `unittest-prereqs` target here? And in that case the three local variables `[UNITTEST|FUNCTIONALTEST|BENCHMARK]_PREREQS` are not actually needed, the lists of prereqs can be inlined in the `add_custom_target` calls.